server/segment_rpc_test.go

```diff
@@ -779,18 +779,55 @@ func TestSubmitSegment_GenPaymentError(t *testing.T) {
 	expErr := errors.New("CreateTicketBatch error")
 	sender.On("CreateTicketBatch", mock.Anything, mock.Anything).Return(nil, expErr)
 
+	oInfo := &net.OrchestratorInfo{
+		PriceInfo: &net.PriceInfo{
+			PricePerUnit:  0,
+			PixelsPerUnit: 1,
+		},
+	}
+
 	s := &BroadcastSession{
-		Broadcaster: b,
-		ManifestID:  core.RandomManifestID(),
-		Sender:      sender,
-		Balance:     balance,
+		Broadcaster:      b,
+		ManifestID:       core.RandomManifestID(),
+		Sender:           sender,
+		Balance:          balance,
+		OrchestratorInfo: oInfo,
 	}
 
 	_, err := SubmitSegment(s, &stream.HLSSegment{}, 0)
 
 	assert.EqualError(t, err, expErr.Error())
 }
 
+func TestSubmitSegment_GenPaymentError_ValidatePriceError(t *testing.T) {
+	b := stubBroadcaster2()
+	balance := &mockBalance{}
+	balance.On("StageUpdate", mock.Anything, mock.Anything).Return(1, nil, nil)
+	sender := &pm.MockSender{}
+	sender.On("EV", mock.Anything).Return(big.NewRat(1, 1), nil)
+
+	oinfo := &net.OrchestratorInfo{
+		PriceInfo: &net.PriceInfo{
+			PricePerUnit:  1,
+			PixelsPerUnit: 3,
+		},
+	}
+
+	s := &BroadcastSession{
+		Broadcaster:      b,
+		ManifestID:       core.RandomManifestID(),
+		Sender:           sender,
+		Balance:          balance,
+		OrchestratorInfo: oinfo,
+	}
+
+	BroadcastCfg.SetMaxPrice(big.NewRat(1, 5))
+
+	_, err := SubmitSegment(s, &stream.HLSSegment{}, 0)
+
+	assert.Error(t, err)
```

Comment:
Check that this is the error we expect?

Reply:
fixed in https://github.com/livepeer/go-livepeer/pull/992/commits/18648d78c0d532e7e1cc26d8bc1e374a560cee34